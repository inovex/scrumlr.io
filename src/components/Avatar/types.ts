export const AVATAR_SKIN_COLORS = ["Tanned", "Yellow", "Pale", "Light", "Brown", "DarkBrown", "Black"] as const;
export type AvatarSkinColor = (typeof AVATAR_SKIN_COLORS)[number];

export const AVATAR_TOP_TYPES = [
  "Eyepatch",
  "Hat",
  "Hijab",
  "LongHairBigHair",
  "LongHairBob",
  "LongHairBun",
  "LongHairCurly",
  "LongHairCurvy",
  "LongHairDreads",
  "LongHairFrida",
  "LongHairFro",
  "LongHairFroBand",
  "LongHairMiaWallace",
  "LongHairNotTooLong",
  "LongHairShavedSides",
  "LongHairStraight",
  "LongHairStraight2",
  "LongHairStraightStrand",
  "NoHair",
  "ShortHairDreads01",
  "ShortHairDreads02",
  "ShortHairFrizzle",
  "ShortHairShaggyMullet",
  "ShortHairShortCurly",
  "ShortHairShortFlat",
  "ShortHairShortRound",
  "ShortHairShortWaved",
  "ShortHairSides",
  "ShortHairTheCaesar",
  "ShortHairTheCaesarSidePart",
  "Turban",
  "WinterHat1",
  "WinterHat2",
  "WinterHat3",
  "WinterHat4",
] as const;
export type AvatarTopType = (typeof AVATAR_TOP_TYPES)[number];

export const AVATAR_CLOTHE_COLORS = [
  "Black",
  "Blue01",
  "Blue02",
  "Blue03",
  "Gray01",
  "Gray02",
  "Heather",
  "PastelBlue",
  "PastelGreen",
  "PastelOrange",
  "PastelRed",
  "PastelYellow",
  "Pink",
  "Red",
  "White",
] as const;
export type AvatarClotheColor = (typeof AVATAR_CLOTHE_COLORS)[number];

export const AVATAR_GRAPHIC_TYPES = ["Bat", "Cumbia", "Deer", "Diamond", "Hola", "Pizza", "Resist", "Selena", "Bear", "SkullOutline", "Skull"] as const;
export type AvatarGraphicType = (typeof AVATAR_GRAPHIC_TYPES)[number];

export const AVATAR_CLOTHE_TYPES = ["BlazerShirt", "BlazerSweater", "CollarSweater", "GraphicShirt", "Hoodie", "Overall", "ShirtCrewNeck", "ShirtScoopNeck", "ShirtVNeck"] as const;
export type AvatarClotheType = (typeof AVATAR_CLOTHE_TYPES)[number];

export const AVATAR_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "PastelPink", "Blue", "Platinum", "Red", "SilverGray"] as const;
export type AvatarHairColor = (typeof AVATAR_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "Platinum", "Red"] as const;
export type AvatarFacialHairColor = (typeof AVATAR_FACIAL_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_TYPES = ["Blank", "BeardMedium", "BeardLight", "BeardMajestic", "MoustacheFancy", "MoustacheMagnum"] as const;
export type AvatarFacialHairType = (typeof AVATAR_FACIAL_HAIR_TYPES)[number];

export const AVATAR_ACCESSORIES_TYPES = ["Blank", "Kurt", "Prescription01", "Prescription02", "Round", "Sunglasses", "Wayfarers"] as const;
export type AvatarAccessoriesType = (typeof AVATAR_ACCESSORIES_TYPES)[number];

export const AVATAR_EYE_TYPES = ["Close", "Cry", "Default", "Dizzy", "EyeRoll", "Happy", "Hearts", "Side", "Squint", "Surprised", "Wink", "WinkWacky"] as const;
export type AvatarEyeType = (typeof AVATAR_EYE_TYPES)[number];

export const AVATAR_EYEBROW_TYPES = [
  "Angry",
  "AngryNatural",
  "Default",
  "DefaultNatural",
  "FlatNatural",
  "FrownNatural",
  "RaisedExcited",
  "RaisedExcitedNatural",
  "SadConcerned",
  "SadConcernedNatural",
  "UnibrowNatural",
  "UpDown",
  "UpDownNatural",
] as const;
export type AvatarEyebrowType = (typeof AVATAR_EYEBROW_TYPES)[number];

export const AVATAR_MOUTH_TYPES = ["Concerned", "Default", "Disbelief", "Eating", "Grimace", "Sad", "ScreamOpen", "Serious", "Smile", "Tongue", "Twinkle", "Vomit"] as const;
export type AvatarMouthType = (typeof AVATAR_MOUTH_TYPES)[number];
