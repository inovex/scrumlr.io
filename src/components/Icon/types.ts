export type IconNames =
  | 'phase1-create'
  | 'phase1-communicate'
  | 'phase1-combine'
  | 'phase2-vote'
  | 'phase3-discuss'
  | 'phase3-define'
  | 'circle-arrow-left'
  | 'circle-arrow-right'
  | 'circle-add'
  | 'circle'
  | 'more'
  | 'plus'
  | 'circle-selection'
  | 'circle-selection-grey'
  | 'logout'
  | 'download'
  | 'edit'
  | 'donate'
  | 'pencil'
  | 'focus'
  | 'close'
  | 'close20'
  | 'close-circle'
  | 'feedback'
  | 'key-up'
  | 'key-down'
  | 'key-esc'
  | 'settings'
  | 'board'
  | 'trash'
  | 'chevron-left'
  | 'chevron-right'
  | 'stack-top'
  | 'stack-mid'
  | 'stack-hover'
  | 'arrow-left'
  | 'arrow-right';
