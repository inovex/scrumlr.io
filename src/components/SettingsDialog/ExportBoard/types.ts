import {Color} from "constants/colors";

export interface BoardType {
  id: string;
  name: string;
  showAuthors: boolean;
  showVoting: string;
  accessPolicy: string;
}

export interface ColumnType {
  id: string;
  name: string;
  color: Color;
  visible: boolean;
}

export interface NoteType {
  id: string;
  author: string;
  text: string;
  position: {
    column: string;
    stack: string | null;
    rank: number;
  };
}

export interface ParticipantType {
  role: string;
  user: {
    id: string;
    name: string;
  };
}

export interface VotingType {
  votes: {
    votes: number;
    votesPerNote: [];
  };
}

export interface BoardDataType {
  board: BoardType;
  columns: ColumnType[];
  notes: NoteType[];
  participants: ParticipantType[];
  votings: VotingType[];
}
