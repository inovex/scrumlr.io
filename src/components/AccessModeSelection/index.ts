import AccessModeSelection from './AccessModeSelection';
export default AccessModeSelection;
export * from './AccessModeSelection';
