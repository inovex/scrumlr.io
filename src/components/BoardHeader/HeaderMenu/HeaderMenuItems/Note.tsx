import store, {useAppSelector} from "store";
import {ApplicationState} from "types/store";
import {ActionFactory} from "store/action";
import "./HeaderMenuItems.scss";
import classNames from "classnames";

export type NoteProps = {};

export const Note = (props: NoteProps) => {
  const state = useAppSelector((applicationState: ApplicationState) => ({
    board: applicationState.board.data!,
  }));

  return (
    <div className="menu__item-button" id="note">
      <li className="header-menu__item">
        <button
          className="menu__item-button"
          onClick={() => {
            store.dispatch(ActionFactory.editBoard({id: state.board!.id, showNotesOfOtherUsers: !state.board!.showNotesOfOtherUsers}));
          }}
        >
          <div className="item-button__toggle-container">
            <div
              className={classNames(
                "item-button__toggle",
                {"item-button__toggle--left": state.board!.showNotesOfOtherUsers},
                {"item-button__toggle--right": !state.board!.showNotesOfOtherUsers}
              )}
            />
          </div>
          <label className="item-button__label">{state.board!.showNotesOfOtherUsers ? "Hide" : "Show"} notes of other users</label>
        </button>
      </li>
    </div>
  );
};
