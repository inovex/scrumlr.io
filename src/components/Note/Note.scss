@import "constants/style";

$note__border-radius: 16px;
$note__text-margin-bottom: 9px;
$note__box-shadow-color: rgba(0, 87, 255, 0.16);
$note__box-shadow-color-dark-mode: rgba(0, 0, 0, 0.2);

$note__indicator-height: 34px;
$note__indicator-width: 3px;

$note__author-name-size: 12px;
$note__author-image-size: 26px;
$note__author-margin-left: 10px;

.note {
  padding: $padding--default $padding--medium;
  position: relative;
  border-radius: $note__border-radius;
  background-color: $color-white;
  box-shadow: 0 6px 9px 0 $note__box-shadow-color;
  transition: transform 0.3s ease-in-out, box-shadow 0.3s ease-in-out;
  list-style-type: none;
  z-index: $note-z-index;
}
@media (prefers-color-scheme: dark) {
  .note {
    background-color: $color-dark-mode-note;
    box-shadow: 0 6px 9px 0 $note__box-shadow-color-dark-mode;
  }
}
.note:hover {
  transform: scale(1.02);
}

.note--own-card::before {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  width: $note__indicator-width;
  height: $note__indicator-height;
  border-radius: 0 $note__indicator-width $note__indicator-width 0;
  background: var(--accent-color);
}

.note__text {
  margin: 0;
  min-height: 3 * $line-height--medium;
  color: $color-black;
  font-size: $text-size;
  letter-spacing: $letter-spacing--medium;
  line-height: $line-height--medium;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  margin-bottom: $note__text-margin-bottom;
}
@media (prefers-color-scheme: dark) {
  .note__text {
    color: $color-white;
  }
}

.note__author {
  display: flex;
  align-items: center;
  // reset css from figure-element:
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.note__author-image {
  height: $note__author-image-size;
  width: $note__author-image-size;
  border-radius: 100%;
}

.note__author-name {
  margin-left: $note__author-margin-left;
  color: $color-dark-gray;
  font-size: $note__author-name-size;
  letter-spacing: $letter-spacing--small;
  line-height: $line-height--small;
}
