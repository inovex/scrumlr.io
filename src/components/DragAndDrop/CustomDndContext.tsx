import {
  Collision,
  CollisionDetection,
  defaultDropAnimationSideEffects,
  DndContext,
  DragOverEvent,
  DragOverlay,
  DragStartEvent,
  DropAnimation,
  MouseSensor,
  rectIntersection,
  TouchSensor,
  UniqueIdentifier,
  useSensor,
  useSensors,
} from "@dnd-kit/core";
import classNames from "classnames";
import {Note} from "components/Note";
import {getColorClassName} from "constants/colors";
import {MOVE_THRESHOLD} from "constants/misc";
import {ReactNode, useState} from "react";
import {useAppSelector} from "store";
import {Column} from "types/column";
import {isEqual} from "underscore";

type CustomDndContextProps = {
  children: ReactNode;
};

export const CustomDndContext = ({children}: CustomDndContextProps) => {
  const [maxCollision, setMaxCollision] = useState<Collision | undefined>();
  const [dragActive, setDragActive] = useState<{id: string; colorClassName: string} | undefined>();

  const {self, columns, notes} = useAppSelector((state) => ({self: state.participants?.self, columns: state.columns, notes: state.notes}), isEqual);

  const sensors = useSensors(useSensor(MouseSensor, {activationConstraint: {distance: 2}}), useSensor(TouchSensor, {activationConstraint: {delay: 200, tolerance: 5}}));
  const dropAnimation: DropAnimation = {
    sideEffects: defaultDropAnimationSideEffects({
      styles: {
        active: {
          opacity: "0.5",
        },
      },
    }),
  };

  const isNote = (collision: Collision) => collision.data?.droppableContainer.data.current.type === "note";

  const getColumn = (id?: UniqueIdentifier): Column | undefined => {
    if (!id) return undefined;
    const note = notes.find((n) => n.id === id);
    if (note) return columns.find((column) => column.id === note.position.column);
    return columns.find((column) => column.id === id);
  };

  const onDragStart = ({active}: DragStartEvent) => {
    if (!active) return;
    const activeColumnColor = getColumn(active.id)?.color;
    if (!activeColumnColor) return;
    setDragActive({id: active.id.toString(), colorClassName: getColorClassName(activeColumnColor)});
  };

  const onDragOver = ({collisions}: DragOverEvent) => {
    const overCollision = collisions?.at(0);
    if (!overCollision) return;
    if (isNote(overCollision)) setMaxCollision(overCollision);
  };

  const collisionDetectionWrapper =
    (collisionDetection: CollisionDetection) =>
    (...[args]: Parameters<typeof collisionDetection>) => {
      const collisions = collisionDetection(args);

      const overCollision = collisions.at(0);
      // if highest collision ratio is below threshold, place previous collision in front
      if (maxCollision && overCollision && isNote(overCollision) && overCollision.data?.value < MOVE_THRESHOLD) {
        // find last maxCollision in collisions
        const lastOver = collisions.findIndex((collision) => collision.id === maxCollision?.id);
        if (lastOver === -1) collisions.unshift({...maxCollision, data: {...maxCollision.data, value: 0}});
        else collisions.unshift(...collisions.splice(lastOver, 1));
      }

      return collisions;
    };

  return (
    <DndContext sensors={sensors} onDragStart={onDragStart} onDragOver={onDragOver} collisionDetection={collisionDetectionWrapper(rectIntersection)}>
      {children}

      <DragOverlay zIndex={1000} dropAnimation={dropAnimation} className={classNames("drag-overlay", dragActive?.colorClassName)}>
        {dragActive?.id && self ? <Note noteId={dragActive.id.toString()} viewer={self} /> : null}
      </DragOverlay>
    </DndContext>
  );
};
