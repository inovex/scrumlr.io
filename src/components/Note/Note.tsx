import "./Note.scss";
import avatar from "assets/avatar.png";
import {useSelector} from "react-redux";
import {ApplicationState} from "types/store";
import classNames from "classnames";
import Parse from "parse";

export interface NoteProps {
  text: string;
  authorId: string;
}

const Note = ({text, authorId}: NoteProps) => {
  const state = useSelector((applicationState: ApplicationState) => ({
    board: applicationState.board,
    notes: applicationState.notes,
    users: applicationState.users,
  }));

  return (
    <li className={classNames("note", {"note--own-card": Parse.User.current()?.id === authorId})}>
      <p className="note__text">{text}</p>
      <footer className="note__footer">
        <figure className="note__author" aria-roledescription="author">
          <img className="note__author-image" src={avatar} alt="User" />
          <figcaption className="note__author-name">{state.users.all.filter((user) => user.id === authorId)[0]?.displayName}</figcaption>
        </figure>
      </footer>
    </li>
  );
};
export default Note;
