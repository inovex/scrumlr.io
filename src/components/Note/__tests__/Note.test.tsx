import {fireEvent} from "@testing-library/react";
import {Note} from "components/Note";
import {Provider} from "react-redux";
import {render} from "testUtils";
import getTestStore from "utils/test/getTestStore";
import getTestParticipant from "utils/test/getTestParticipant";
import * as reactRouter from "react-router";
import {ApplicationState} from "types";
import {BoardState} from "types/board";
import getTestApplicationState from "utils/test/getTestApplicationState";
import {CustomDndContext} from "components/DragAndDrop/CustomDndContext";

const NOTE_ID = "test-notes-id-1";

const createBoardData = (overwrite?: Partial<BoardState["data"]> & Partial<Pick<BoardState, "status">>): BoardState => {
  return {
    status: overwrite?.status ?? "ready",
    data: {
      id: overwrite?.id ?? "note-tests-board",
      name: overwrite?.name ?? "note-tests-board",
      accessPolicy: overwrite?.accessPolicy ?? "PUBLIC",
      showAuthors: overwrite?.showAuthors ?? false,
      showNoteReactions: overwrite?.showNoteReactions ?? true,
      showNotesOfOtherUsers: overwrite?.showNotesOfOtherUsers ?? true,
      allowStacking: overwrite?.allowStacking ?? true,
    },
  };
};

const createNote = (isModerator: boolean, overwrite?: Partial<ApplicationState>) => {
  return (
    <Provider store={getTestStore(overwrite)}>
      <CustomDndContext>
        <Note key={NOTE_ID} noteId={NOTE_ID} viewer={getTestParticipant(isModerator ? {role: "MODERATOR"} : {role: "PARTICIPANT"})} />
      </CustomDndContext>
    </Provider>
  );
};

describe("Note", () => {
  beforeEach(() => {
    window.IntersectionObserver = jest.fn(
      () =>
        ({
          observe: jest.fn(),
          disconnect: jest.fn(),
        }) as unknown as IntersectionObserver
    );
  });

  describe("content", () => {
    it("should display the text content of the note", () => {
      const note = getTestApplicationState().notes.find((n) => n.id === NOTE_ID);
      expect(note).toBeDefined();
      const {container} = render(createNote(false));
      expect(container.firstChild).toHaveTextContent(note!.text);
    });
  });

  describe("author information", () => {
    it("should be displayed on enabled author visibility", () => {
      const board = createBoardData({showAuthors: true});
      const {container} = render(createNote(false, {board: board}));
      expect(container.firstChild).toContainHTML("figure");
    });
    it("should not be displayed on disabled author visibility", () => {
      const board = createBoardData({showAuthors: false});
      const {container} = render(createNote(false, {board: board}));
      expect(container.firstChild).not.toContainHTML("figure");
    });
  });

  describe("votes", () => {
    it("should contain votes container element", () => {
      const {container} = render(createNote(false));
      expect(container.getElementsByClassName("votes").length).toBeGreaterThan(0);
    });
  });

  describe("side effects", () => {
    // skip because I can't fix it right now, see: https://stackoverflow.com/q/64445312
    it.skip("should navigate to stack route on click", () => {
      const navigateSpy = jest.fn();
      jest.spyOn(reactRouter, "useNavigate").mockImplementationOnce(() => navigateSpy);
      const {container} = render(createNote(false));
      fireEvent.click(container.querySelector(".note")!);
      expect(navigateSpy).toHaveBeenCalledWith(`note/${NOTE_ID}/stack`);
    });
  });
});
