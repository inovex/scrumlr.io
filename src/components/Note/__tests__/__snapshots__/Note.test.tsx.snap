// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Test NoteDialog created/not created NoteDialog is not present: snapshot 1`] = `
<div
  id="portal"
>
  <li
    class="note__root"
  >
    <div
      class="note"
    >
      <div
        class="note__content"
      >
        <p
          class="note__text"
        >
          Test Text
        </p>
        <svg
          class="note__edit"
        >
          icon-edit.svg
        </svg>
      </div>
      <footer
        class="note__footer"
      >
        <figure
          aria-roledescription="author"
          class="note__author"
        >
          <div
            class="user-avatar note__user-avatar"
          >
            <div
              class="user__initials"
              title=""
            >
              NA
            </div>
          </div>
          <figcaption
            class="note__author-name"
          />
        </figure>
        <div
          class="votes note__votes"
        >
          <button
            class="dot-button dot-button__delete"
          >
            <span
              class="dot-button__folded-corner"
            />
            <span>
              3
            </span>
          </button>
          <button
            class="dot-button dot-button__add"
          >
            <svg
              class="dot-button__add-icon"
            >
              icon-add.svg
            </svg>
          </button>
        </div>
      </footer>
    </div>
    <div
      class="note__in-stack"
    />
  </li>
</div>
`;

exports[`Note Test NoteDialog created/not created NoteDialog is present: snapshot 1`] = `
<div
  id="portal"
>
  <div
    class="portal portal--darkBackground"
    role="dialog"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="portal__frame"
      >
        <div
          class="portal__content"
          role="dialog"
        >
          <div
            class="note-dialog accent-color__ note-dialog__disabled-pointer"
          >
            <h2
              class="note-dialog__header"
            />
            <div
              class="note-dialog__note"
            >
              <div
                class="note-dialog__content"
              >
                <blockquote
                  class="note-dialog__text"
                  contenteditable="false"
                >
                  Test Text
                </blockquote>
              </div>
              <footer
                class="note-dialog__footer"
              >
                <figure
                  class="note-dialog__author"
                >
                  <div
                    class="user-avatar note-dialog__user-avatar"
                  >
                    <div
                      class="user__initials"
                      title=""
                    >
                      NA
                    </div>
                  </div>
                  <figcaption
                    class="note-dialog__author-name"
                  />
                </figure>
                <div
                  class="votes note__votes"
                >
                  <button
                    class="dot-button dot-button__delete"
                  >
                    <span
                      class="dot-button__folded-corner"
                    />
                    <span>
                      3
                    </span>
                  </button>
                  <button
                    class="dot-button dot-button__add"
                  >
                    <svg
                      class="dot-button__add-icon"
                    >
                      icon-add.svg
                    </svg>
                  </button>
                </div>
              </footer>
            </div>
            <div
              class="note-dialog__note"
            >
              <div
                class="note-dialog__content"
              >
                <blockquote
                  class="note-dialog__text"
                  contenteditable="false"
                />
              </div>
              <footer
                class="note-dialog__footer"
              >
                <figure
                  class="note-dialog__author"
                >
                  <img
                    alt="User"
                    class="note-dialog__author-image"
                    src="avatar.png"
                  />
                  <figcaption
                    class="note-dialog__author-name"
                  />
                </figure>
                <div
                  class="votes note__votes"
                >
                  <button
                    class="dot-button dot-button__add"
                  >
                    <svg
                      class="dot-button__add-icon"
                    >
                      icon-add.svg
                    </svg>
                  </button>
                </div>
              </footer>
            </div>
            <div
              class="note-dialog__note"
            >
              <div
                class="note-dialog__content"
              >
                <blockquote
                  class="note-dialog__text"
                  contenteditable="false"
                />
              </div>
              <footer
                class="note-dialog__footer"
              >
                <figure
                  class="note-dialog__author"
                >
                  <img
                    alt="User"
                    class="note-dialog__author-image"
                    src="avatar.png"
                  />
                  <figcaption
                    class="note-dialog__author-name"
                  />
                </figure>
                <div
                  class="votes note__votes"
                >
                  <button
                    class="dot-button dot-button__add"
                  >
                    <svg
                      class="dot-button__add-icon"
                    >
                      icon-add.svg
                    </svg>
                  </button>
                </div>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
  <li
    class="note__root note__root-disabled-click"
  >
    <div
      class="note note__disabled-click"
    >
      <div
        class="note__content"
      >
        <p
          class="note__text"
        >
          Test Text
        </p>
        <svg
          class="note__edit"
        >
          icon-edit.svg
        </svg>
      </div>
      <footer
        class="note__footer"
      >
        <figure
          aria-roledescription="author"
          class="note__author"
        >
          <div
            class="user-avatar note__user-avatar"
          >
            <div
              class="user__initials"
              title=""
            >
              NA
            </div>
          </div>
          <figcaption
            class="note__author-name"
          />
        </figure>
        <div
          class="votes note__votes"
        >
          <button
            class="dot-button dot-button__delete"
          >
            <span
              class="dot-button__folded-corner"
            />
            <span>
              3
            </span>
          </button>
          <button
            class="dot-button dot-button__add"
          >
            <svg
              class="dot-button__add-icon"
            >
              icon-add.svg
            </svg>
          </button>
        </div>
      </footer>
    </div>
    <div
      class="note__in-stack"
    />
  </li>
</div>
`;

exports[`Note should have correct style show note with correct style 1`] = `
<li
  class="note__root"
>
  <div
    class="note"
  >
    <div
      class="note__content"
    >
      <p
        class="note__text"
      >
        Test Text
      </p>
      <svg
        class="note__edit"
      >
        icon-edit.svg
      </svg>
    </div>
    <footer
      class="note__footer"
    >
      <figure
        aria-roledescription="author"
        class="note__author"
      >
        <div
          class="user-avatar note__user-avatar"
        >
          <div
            class="user__initials"
            title=""
          >
            NA
          </div>
        </div>
        <figcaption
          class="note__author-name"
        />
      </figure>
      <div
        class="votes note__votes"
      >
        <button
          class="dot-button dot-button__delete"
        >
          <span
            class="dot-button__folded-corner"
          />
          <span>
            3
          </span>
        </button>
        <button
          class="dot-button dot-button__add"
        >
          <svg
            class="dot-button__add-icon"
          >
            icon-add.svg
          </svg>
        </button>
      </div>
    </footer>
  </div>
  <div
    class="note__in-stack"
  />
</li>
`;
